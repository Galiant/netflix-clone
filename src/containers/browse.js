/*eslint-disable react/prop-types*/
/*eslint-disable no-unused-vars*/
import React, { useContext, useState, useEffect } from 'react';
import { SelectProfileContainer } from './profiles';
import { FirebaseContext } from '../context/firebase';
import { Header, Loading } from '../components';
import * as ROUTES from '../constants/routes';
import logo from '../logo.svg';

export function BrowseContainer({ slides }) {
  const [profile, setProfile] = useState({});
  const [loading, setLoading] = useState(true);
  const { firebase } = useContext(FirebaseContext);
  const user = firebase.auth().currentUser || {};

  useEffect(() => {
    setTimeout(() => {
      setLoading(false);
    }, 3000);
  }, [profile.displayName]);

  return profile.displayName ? (
    <>
      {loading ? <Loading src={user.photoURL} /> : <Loading.ReleaseBody />}

      <Header src='joker1' dontShowOnSmallViewPort>
        <Header.Frame>
          <Header.Group>
            <Header.Logo to={ROUTES.HOME} src={logo} alt='Netflix' />
            <Header.TextLink>Series</Header.TextLink>
            <Header.TextLink>Films</Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.Profile>
              <Header.Picture src={user.photoURL} />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src={user.photoURL} />
                  <Header.TextLink>{user.displayName}</Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
        <Header.Feature>
          <Header.FeatureText>Watch Joker Now</Header.FeatureText>
          <Header.Text>
            Joker is a 2019 American psychological thriller film directed and
            produced by Todd Phillips, who co-wrote the screenplay with Scott
            Silver. The film, based on DC Comics characters, stars Joaquin
            Phoenix as the Joker and provides an alternative origin story for
            the character.
          </Header.Text>
        </Header.Feature>
      </Header>
    </>
  ) : (
    <SelectProfileContainer user={user} setProfile={setProfile} />
  );
}
