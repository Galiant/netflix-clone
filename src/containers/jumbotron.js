import React from 'react';
import { Jumbotron } from '../components';
import jumbotronData from '../fixtures/jumbotron.json';

function JumbotronContainer() {
  return (
    <Jumbotron.Container>
      {jumbotronData.map(item => (
        <Jumbotron key={item.id} direction={item.direction}>
          <Jumbotron.Pane>
            <Jumbotron.Title>{item.title}</Jumbotron.Title>
            <Jumbotron.SubTitle>{item.subTitle}</Jumbotron.SubTitle>
          </Jumbotron.Pane>
          <Jumbotron.Pane>
            <Jumbotron.Image
              src={item.image}
              alt={item.alt}
              data-testid={`${item.id}-jumbo-image`}
            />
          </Jumbotron.Pane>
        </Jumbotron>
      ))}
    </Jumbotron.Container>
  );
}

export default JumbotronContainer;
