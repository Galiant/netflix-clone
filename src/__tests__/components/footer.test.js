import React from 'react';
import { render } from '@testing-library/react';
import { FooterContainer } from '../../containers/footer';

describe('Footer', () => {
  it('renders the Footer with populated data', () => {
    const { container, getByText } = render(<FooterContainer />);

    expect(getByText('Questions? Contact us.')).toBeTruthy();
    expect(getByText('FAQ')).toBeTruthy();
    expect(getByText('Investor Relations')).toBeTruthy();
    expect(getByText('Ways To Watch')).toBeTruthy();
    expect(getByText('Corporate Information')).toBeTruthy();
    expect(getByText('Netflix Originals')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
