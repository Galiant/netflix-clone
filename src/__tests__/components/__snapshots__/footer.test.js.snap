// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the Footer with populated data 1`] = `
<div
  class="footer__Container-eh6b4z-0 kpqbmQ"
>
  <p
    class="footer__Title-eh6b4z-4 dQKxha"
  >
    Questions? Contact us.
  </p>
  <p
    class="footer__Break-eh6b4z-6 hHvrrV"
  />
  <div
    class="footer__Row-eh6b4z-2 khVTlG"
  >
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Ways To Watch
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Help Center
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Reedem Gift Cards
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Media Center
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Buy Gift Cards
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-eh6b4z-3 kvyrnZ"
        href="#"
      >
        Legal Notices
      </a>
    </div>
  </div>
  <p
    class="footer__Break-eh6b4z-6 hHvrrV"
  />
  <p
    class="footer__Text-eh6b4z-5 jhiiHr"
  >
    Netflix Ireland
  </p>
</div>
`;
